src/System.Data.SqlClient/src/System/Data/SqlClient/SqlDelegatedTransaction.cs

```diff
@@ -461,10 +461,14 @@ private void ValidateActiveOnConnection(SqlInternalConnection connection)
         // Skip first 4 bytes since they contain the version
         private Guid GetGlobalTxnIdentifierFromToken()
         {
+#if netcoreapp
```

Comment:
Outside of SQlClient we try to avoid `#if netcoreapp` because it makes the code harder to read. This change nearly doubles the number of them in SqlClient

Admittedly most of the rest of the repo is not shipping different implementation for multiple platforms, which SqlClient has to do.

Reply:
I don't know anything about the scenario, but perhaps the perf gain does not justify the complexity.

Reply:
In other cases i've put functionality into *.netcore.cs or *.netstandard.cs files as directed. I could implement platform specific function for each of these but it didn't seem that putting tiny parts of a 10000 line file into separate files helped readability. If it isn't justified or needs to be done another way when reviewed that's fine.